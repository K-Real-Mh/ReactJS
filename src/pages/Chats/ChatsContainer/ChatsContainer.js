import { Redirect, useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { chatListSelector } from '../../../store/chats/selectors';
import { useEffect, useMemo } from 'react';
import { messageListSelector } from '../../../store/messages/selectors';
import Chats from '../Chats';
import { initMessageTracking } from '../../../store/messages/actions';

export default function ChatsContainer() {
    const dispatch = useDispatch();
    const {chatId} = useParams();

    const getMessageSelected = useMemo(() => {
        return messageListSelector(chatId)
    }, [chatId]);

    const messages = useSelector(
        getMessageSelected,
        (prev, next) => prev.length === next.length
    );

    useEffect(() => {
        dispatch(initMessageTracking())
    }, [])

    const chats = useSelector(
        chatListSelector,
        (prev, next) => prev.length === next.length
    );

    if (!chatId || !messages) {
        return <Redirect to="/nochat"/>;
    }

    return (
        <Chats
            chats={chats}
            chatId={chatId}
            messages={messages}
        />
    );
}
