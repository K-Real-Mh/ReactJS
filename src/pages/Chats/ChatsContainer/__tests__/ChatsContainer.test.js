import { render } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store'
import ChatsContainer from '../ChatsContainer';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('ChatsContainer tests', function () {
    it('should matches snapshot', function () {

        const initialState = {
            messages: {
                messageList: {
                    id0: [
                        {
                            id: 'id00',
                            text: 'asdaaaa',
                            author: 'you'
                        },
                        {
                            id: 'id01',
                            text: 'hello, my friend!',
                            author: 'Han Solo'
                        },
                    ],
                    id1: [
                        {
                            id: 'id00',
                            text: 'hello !',
                            author: 'you'
                        },
                        {
                            id: 'id01',
                            text: 'hello, my friend!',
                            author: 'Han Solo'
                        },
                    ],
                }
            },
            chats: {
                chatList: [
                    {
                        id: 'id0',
                        name: 'Han Solo'
                    },
                    {
                        id: 'id1',
                        name: 'Vasya'
                    },
                ],
            }

        }
        const store = mockStore(initialState)

        const component = render(
            <Provider store={store}>
                <MemoryRouter initialEntries={['chats/id0']}>
                    <Route path="chats/:chatId">
                        <ChatsContainer/>
                    </Route>
                </MemoryRouter>
            </Provider>
        );


        expect(component).toMatchSnapshot();
    });
});
