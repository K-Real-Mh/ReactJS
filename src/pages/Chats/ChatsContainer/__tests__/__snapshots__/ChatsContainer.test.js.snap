// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatsContainer tests should  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <nav
            class="MuiList-root list MuiList-padding"
          >
            <a
              href="/chats/id0"
            >
              <div
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button Mui-selected"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    Han Solo
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              href="/chats/id1"
            >
              <div
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    Vasya
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </a>
          </nav>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
        >
          <p
            class="Message"
          >
            you: 
            asdaaaa
          </p>
          <p
            class="Message"
          >
            Han Solo: 
            hello, my friend!
          </p>
          <form
            autocomplete="off"
            novalidate=""
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
                for="standard-textarea"
                id="standard-textarea-label"
              >
                Message
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
              >
                <textarea
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                  id="standard-textarea"
                  rows="1"
                  style="height: 0px; overflow: hidden;"
                />
                <textarea
                  aria-hidden="true"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                  readonly=""
                  style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                  tabindex="-1"
                />
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Send
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <nav
          class="MuiList-root list MuiList-padding"
        >
          <a
            href="/chats/id0"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button Mui-selected"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Han Solo
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </a>
          <a
            href="/chats/id1"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Vasya
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </a>
        </nav>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <p
          class="Message"
        >
          you: 
          asdaaaa
        </p>
        <p
          class="Message"
        >
          Han Solo: 
          hello, my friend!
        </p>
        <form
          autocomplete="off"
          novalidate=""
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
              for="standard-textarea"
              id="standard-textarea-label"
            >
              Message
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                id="standard-textarea"
                rows="1"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                tabindex="-1"
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Send
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
